docs/data/release_archive.yml

```diff
@@ -490,6 +490,11 @@ release_archive:
       version: 3.0.1
       release_date: 2023-05-09
       filename: "gcp-pubsub"
+    - name: "Flink Pulsar Connector"
```

Comment:
NIT: should we keep the connectors sorted alphabetically? It will be easier for devs to maintain. WDYT?

Reply:
It's order by release date, I think it‘s fine to keep current order.